hack/update-vendor.sh

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+# Explicitly clear GOFLAGS, since GOFLAGS=-mod=vendor breaks dependency resolution while rebuilding vendor
```

Comment:
Can you elaborate on this? In my team's projects which use modules with vendor enabled, we have no choice but to  use `GOFLAGS=-mod=vendor` for all our tooling to work, and we've had no issues with rebuilding vendor. So I just wonder whether this is idiosyncratic to origin's build (in which case maybe that should be detailed here) or whether you were alluding to some Go issue of which I was not (but maybe should be) aware.

Reply:
this is copied from upstream. We might want to clean up that stuff around the actually algorithm.

Reply:
This particular script's job is to build vendor. It should not use the vendor dir while doing so. 

All other scripts can use -mod=vendor